qutip/visualization.py

```diff
@@ -332,38 +383,35 @@ def color_fn(w):
                 _x - 0.5, height - _y + 0.5, W[y, x], w_max,
                 min(1, abs(W[y, x]) / w_max), color_fn=color_fn, ax=ax)
 
-    # color axis
-    vmax = np.pi if color_style == "phase" else abs(W).max()
-    norm = mpl.colors.Normalize(-vmax, vmax)
-    cax, kw = mpl.colorbar.make_axes(ax, shrink=0.75, pad=.1)
-    mpl.colorbar.ColorbarBase(cax, norm=norm, cmap=cmap)
+    if colorbar:
+        vmax = np.pi if color_style == "phase" else abs(W).max()
+        norm = mpl.colors.Normalize(-vmax, vmax)
+        cax, kw = mpl.colorbar.make_axes(ax, shrink=0.75, pad=.1)
+        mpl.colorbar.ColorbarBase(cax, norm=norm, cmap=cmap)
+
+    # axis
+    if not (x_basis or y_basis):
+        ax.axis('off')
+    ax.axis('equal')
+    ax.set_frame_on(False)
 
-    xtics = 0.5 + np.arange(width)
     # x axis
-    ax.xaxis.set_major_locator(plt.FixedLocator(xtics))
-    if xlabels:
-        nxlabels = len(xlabels)
-        if nxlabels != len(xtics):
-            raise ValueError(f"got {nxlabels} xlabels but needed {len(xtics)}")
-        ax.set_xticklabels(xlabels)
-        if label_top:
-            ax.xaxis.tick_top()
-    ax.tick_params(axis='x', labelsize=14)
+    xticks = 0.5 + np.arange(width)
+    if x_basis:
+        _set_ticklabels(ax, x_basis, xticks, 'x')
+    if label_top:
+        ax.xaxis.tick_top()
 
     # y axis
-    ytics = 0.5 + np.arange(height)
-    ax.yaxis.set_major_locator(plt.FixedLocator(ytics))
-    if ylabels:
-        nylabels = len(ylabels)
-        if nylabels != len(ytics):
-            raise ValueError(f"got {nylabels} ylabels but needed {len(ytics)}")
-        ax.set_yticklabels(list(reversed(ylabels)))
-    ax.tick_params(axis='y', labelsize=14)
+    yticks = 0.5 + np.arange(height)
+    if y_basis:
+        _set_ticklabels(ax, list(reversed(y_basis)), yticks, 'y')
 
     return fig, ax
 
 
-def sphereplot(theta, phi, values, fig=None, ax=None, save=False):
+def sphereplot(theta, phi, values, *,
```

Comment:
`theta` and `phi` should range between `[0, pi]` and `[0, 2 pi]`.
It is not clear in the documentation but other range create bad plots.
Could you add a comment for this in the documentation.